src/Lti/Factory/Lti1p3RequestFactory.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/**
+ *  This program is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU General Public License
+ *  as published by the Free Software Foundation; under version 2
+ *  of the License (non-upgradable).
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ *  Copyright (c) 2020 (original work) Open Assessment Technologies S.A.
+ */
+
+declare(strict_types=1);
+
+namespace OAT\SimpleRoster\Lti\Factory;
+
+use OAT\Library\Lti1p3Core\Message\Launch\Builder\LtiResourceLinkLaunchRequestBuilder;
+use OAT\Library\Lti1p3Core\Message\Payload\Claim\ContextClaim;
+use OAT\Library\Lti1p3Core\Registration\RegistrationRepositoryInterface;
+use OAT\Library\Lti1p3Core\Resource\LtiResourceLink\LtiResourceLink;
+use OAT\SimpleRoster\DataTransferObject\LoginHintDto;
+use OAT\SimpleRoster\Entity\Assignment;
+use OAT\SimpleRoster\Lti\Exception\RegistrationNotFoundException;
+use OAT\SimpleRoster\Lti\Request\LtiRequest;
+
+class Lti1p3RequestFactory implements LtiRequestFactoryInterface
+{
+    /** @var RegistrationRepositoryInterface */
+    private $repository;
+
+    /** @var LtiResourceLinkLaunchRequestBuilder */
+    private $builder;
+
+    public function __construct(
+        RegistrationRepositoryInterface $repository,
+        LtiResourceLinkLaunchRequestBuilder $builder
+    ) {
+        $this->repository = $repository;
+        $this->builder = $builder;
+    }
+
+    public function create(Assignment $assignment): LtiRequest
+    {
+        $registrationId = 'demo';
+        $resourceLink = new LtiResourceLink($assignment->getLineItem()->getUri());
+        $registration = $this->repository->find($registrationId);
+
+        if (!$registration) {
+            throw new RegistrationNotFoundException(sprintf('Registration %s not found.', $registrationId));
+        }
+
+        $loginHint = new LoginHintDto(
+            $assignment->getUser()->getUsername(),
+            $assignment->getUser()->getGroupId(),
+            $assignment->getLineItem()->getSlug()
+        );
+
+        $message = $this->builder->buildLtiResourceLinkLaunchRequest(
```

Comment:
Don't we have parameters missing?
because we're sending all of this on 1.1, but not here:
```
		             'lti_message_type' => LtiRequest::LTI_MESSAGE_TYPE,
             'lti_version' => LtiRequest::LTI_VERSION_1P0,
             'context_id' => $this->loadBalancer->getLtiRequestContextId($assignment),
             'roles' => LtiRequest::LTI_ROLE,
             'user_id' => $assignment->getUser()->getUsername(),
             'lis_person_name_full' => $assignment->getUser()->getUsername(),
             'resource_link_id' => $assignment->getId(),
             'lis_outcome_service_url' => $this->router->generate(
                 'updateLtiOutcome',
                 [],
                 UrlGeneratorInterface::ABSOLUTE_URL
             ),
             'lis_result_sourcedid' => $assignment->getId(),
             'launch_presentation_return_url' => $this->ltiLaunchPresentationReturnUrl,
             'launch_presentation_locale' => $this->ltiLaunchPresentationLocale,
         ];
```

This document is doing the comparative:
https://www.imsglobal.org/spec/lti/v1p3/migr
Maybe it'd be good just double check it. :)

Reply:
Not sure what different did you spotted, can you please be more specific? I believe for 1.3 all data goes on the "claims" section.

http://www.imsglobal.org/spec/lti/v1p3/#message-claims

In this case, the lti1p3-bundle already create the necessary claims. We are just adding the context claim

Reply:
There are no parameters in case of LTI 1.3 launch, that's why they are not here, they are needed only for LTI 1.1